### PR TITLE
Added Documentation for How To Run Custom Sandbox Image 

### PR DESCRIPTION
**Pull request by sheunaluko**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?** 

This provides documentation in the form of a markdown file which explains how to create and run a custom docker sandbox for OpenDevin. 

**Give a brief summary of what the PR does, explaining any non-trivial design decisions**

The pull request simply adds one single markdown file. 

**Other references**

n/a

### CODE REVIEW

File: docs/modules/usage/custom_sandbox_guide.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+The default OpenDevin sandbox comes with a [minimal ubuntu configuration](https://github.com/OpenDevin/OpenDevin/blob/main/containers/sandbox/Dockerfile). Your use case may need additional software installed by default. This guide will teach you how to accomplish this by utilizing a custom docker image. 
+
+## Setup
+
+To get started running with your own Docker Sandbox image you need to ensure you can build OpenDevin locally via the following: 
+1. Clone the OpenDevin github repository to your local machine 
+2. In the root (OpenDevin/)  directory, run ```make build```
+3. Then run ```make run```  
+4. Finally navigate your browser to ```localhost:3001``` to ensure that your local build of OpenDevin is functional 
+
```
Comment (xingyaoww) [inline docs/modules/usage/custom_sandbox_guide.md:12]:
```suggestion
Please refer to [Development.md](https://github.com/OpenDevin/OpenDevin/blob/main/Development.md) for more installation details.
```

File: docs/modules/usage/custom_sandbox_guide.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+[core]
+workspace_base="./workspace"
+persist_sandbox=false
+run_as_devin=true
+sandbox_container_image="custom_image"
+sandbox_user_id="1001"
+```
+
+### Port use errors 
+If you see an error about a port being in use or unavailable, try deleting all running Docker Containers and then re-running ```make run```
```
Comment (xingyaoww) [inline docs/modules/usage/custom_sandbox_guide.md:104]:
```suggestion
If you see an error about a port being in use or unavailable, try deleting all running Docker Containers (run `docker ps` and `docker rm` relevant containers) and then re-running ```make run``` 
```

**Review by xingyaoww — APPROVED** [review]
Overall LGTM!! Thanks for contributing this!

**Comment by enyst** [comment]
This is awesome, thank you!

**Comment by xingyaoww** [comment]
Are the suggested changes merged yet? If not, I can start a new PR to add them :)

**Comment by iFurySt** [comment]
Ah, I’m sorry for not noticing that the two suggestions hadn't been revised before they were merged 😅

**Comment by tobitege** [comment]
> Are the suggested changes merged yet? If not, I can start a new PR to add them :)

I don't see any commit after your suggestions, looks like you can open a new PR. 😃

**Pull request merged**